components/gitpod-cli/cmd/tasks.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+	"google.golang.org/grpc"
+)
+
+// tasksCmd represents the tasks command
+var tasksCmd = &cobra.Command{
+	Use:   "tasks",
+	Short: "Interact with workspace tasks",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			cmd.Help()
+		}
+	},
+}
+
+var attachTaskCmdOpts struct {
+	Interactive bool
+	ForceResize bool
+}
+
+// TODO(andreafalzetti): refactor tasks.go to only keep the tasks cmd initialisation, moving the subcommands in their own file
+
+// This func was copied from https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/cmd/terminal.go#L29
+// TODO(andreafalzetti): move it somewhere else so that it's reusable from other cmds
+func dialSupervisor() *grpc.ClientConn {
+	// TODO(andreafalzetti): figure out how to import peroperly supervisor.GetConfig()
+	// cfg, err := supervisor.GetConfig()
+	// if err != nil {
+	// 	log.WithError(err).Fatal("cannot get config")
+	// }
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	url := fmt.Sprintf("localhost:%d", 22999)
+	conn, err := grpc.DialContext(ctx, url, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		log.WithError(err).Fatal("cannot connect to supervisor")
+	}
+
+	// TODO(cw): devise some means to properly close the connection
+
+	return conn
+}
+
+// listTasksCmd represents the tasks list command
+var listTasksCmd = &cobra.Command{
+	Use:   "list",
+	Short: "Lists the workspace tasks and their status",
+	Run: func(cmd *cobra.Command, args []string) {
+		var (
+			client = api.NewStatusServiceClient(dialSupervisor())
+		)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		// TODO(andreafalzetti): ask how to opt-out from the stream! {Observe: false} gives me a stream 😢
+		listen, err := client.TasksStatus(ctx, &api.TasksStatusRequest{Observe: false})
+		if err != nil {
+			fmt.Println("Cannot list tasks")
+		}
+
+		errchan := make(chan error, 5)
+		func() {
+			for {
+				resp, err := listen.Recv()
+				if err != nil {
+					errchan <- err
+				}
+
+				tasks := resp.GetTasks()
+
+				if tasks != nil {
+					// TODO(andreafalzetti): refactor this fmt.Println with https://github.com/rodaine/table
+					for _, task := range tasks {
+						fmt.Println(task.Id, task.Presentation.Name, task.State)
+					}
+				} else {
+					break
+				}
+			}
+		}()
+	},
+}
+
+// attachTaskCmd represents the attach task command
+var attachTaskCmd = &cobra.Command{
```

Comment:
Oh, i should have been clear on the issue.
Attach should attach to a terminal and allow a user to control a task. Like it is done for internal CLI: https://github.com/andreafalzetti/gitpod/blob/d70eb530702440ac8c0eb91dd30270eebe7e9c1d/components/supervisor/cmd/terminal-attach.go#L29